src/main.rs

```diff
@@ -778,7 +784,11 @@ fn run(arg_matches: ArgMatches) -> Result<(), String> {
     );
 
     let runtime = Runtime::new().map_err(|e| format!("could not construct runtime: {}", e))?;
-    let _ = runtime.block_on(main_task(arg_matches, &log, &root_logger))?;
+    while runtime.block_on(main_task(arg_matches.clone(), &log, &root_logger))?
+        == ExitSignal("SIG_HUP")
+    {
+        info!(log, "Received SIG_HUP, restarting");
```

Comment:
This will make it sit in the while loop or what am I missing?

Reply:
it block until it receives a signal. If the signal received is HUP, it will restart, otherwise the loop ends and the program ends.

Reply:
does it actually restart or does it just exit so k8s or another orchestrator restarts it?

Reply:
If you use HUP it will actually restart. If you interrupt or terminate it, then it will be restarted if you indeed have such an orchestrator.